Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -305,14 +399,10 @@ private extension UInt8 {
         case .keepAlive:
             return true
         case .unknown:
-            for header in self.headers {
-                if self.buffer.equalCaseInsensitiveASCII(view: "connection".utf8, at: header.name) {
-                    if self.buffer.equalCaseInsensitiveASCII(view: "close".utf8, at: header.value) {
-                        return false
-                    }
-                    return self.buffer.equalCaseInsensitiveASCII(view: "keep-alive".utf8, at: header.value)
-                }
-            }
+            let keepAliveStatus = self.isKeepAlive
+            if keepAliveStatus == .keepAlive { return true }
+            if keepAliveStatus == .close { return false }
```

Comment:
May as well use the `enum`:

```swift
case .unknown:
    switch self.keepAliveFromHeaders {
    case .keepAlive:
        return true
    case .close:
        return false
    case .unspecified:
        // HTTP 1.1 use keep-alive by default if not otherwise told
        return version.major == 1 && version.minor >= 1
    }
}
```